docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_delivery_failing.mdx

```diff
@@ -23,6 +23,243 @@ CBP
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. **Install and Configure AWS CLI**: Before you can start using AWS CLI, you need to install it on your local system. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. **List Trails**: The first step to check if log files are being delivered without any failures is to list all the trails in your AWS account. You can do this by running the following command: `aws cloudtrail describe-trails`. This command will return a list of all the trails in your account.
+
+3. **Get Trail Status**: For each trail returned by the previous command, you need to check its status. You can do this by running the following command: `aws cloudtrail get-trail-status --name <trail_name>`. Replace `<trail_name>` with the name of the trail you want to check. This command will return the status of the specified trail.
+
+4. **Check Log File Validation**: In the output of the previous command, look for the `LogFileValidationEnabled` field. If its value is `true`, it means that log file validation is enabled for the trail, which is a good sign. However, if its value is `false`, it means that log file validation is not enabled, which could lead to log files being delivered with failures.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. **Import necessary libraries and establish a session**: To start with, you need to import the necessary libraries in your Python script. Boto3 is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python, which allows Python developers to write software that makes use of AWS services like Amazon S3, Amazon EC2, etc. After importing the necessary libraries, establish a session using your AWS credentials.
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+2. **Create a CloudTrail client**: After establishing a session, create a CloudTrail client. This client will help you interact with the CloudTrail service and fetch the necessary details.
+
+```python
+cloudtrail_client = session.client('cloudtrail')
+```
+
+3. **Fetch Trails and Check Log File Validation**: Now, fetch all the trails and check if the log file validation is enabled or not. If the log file validation is not enabled, it means there is a misconfiguration.
+
+```python
+try:
+    response = cloudtrail_client.describe_trails()
+    for trail in response['trailList']:
+        if not trail['LogFileValidationEnabled']:
+            print(f"Log file validation is not enabled for the trail: {trail['Name']}")
+except ClientError as e:
+    print(f"Unexpected error: {e}")
+```
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>